src/codegen/riscv64/macro-assembler-riscv64.h

```diff
@@ -522,13 +522,20 @@ class V8_EXPORT_PRIVATE TurboAssembler : public TurboAssemblerBase {
   void Popcnt32(Register rd, Register rs);
   void Popcnt64(Register rd, Register rs);
 
-  void Ext32(Register rt, Register rs, uint16_t pos, uint16_t size);
-  void Ext64(Register rt, Register rs, uint16_t pos, uint16_t size);
-  void Ins32(Register rt, Register rs, uint16_t pos, uint16_t size);
-  void Ins64(Register rt, Register rs, uint16_t pos, uint16_t size);
-  void ExtractBits(Register dest, Register source, Register pos, int size,
-                   bool sign_extend = false);
-  void InsertBits(Register dest, Register source, Register pos, int size);
+  // Extract bits [pos, pos+size) from 32-bit word of rs to bits [0, size) of
+  // rt; higher bits of rt are set to zero
+  void ExtractBits32(Register rt, Register rs, uint16_t pos, uint16_t size);
+  // Extract bits [pos, pos+size) of rs to bits [0, size) of rt; higher bits
+  // of rt are set to zero
+  void ExtractBits64(Register rt, Register rs, uint16_t pos, uint16_t size);
+  // Extract bits [pos, pos+size) of rs to bits [0, size) of rt; higher bits
+  // of rt are zero- or sign-extended depending on sign_extend
+  void ExtractBits64(Register dest, Register source, Register pos, int size,
```

Comment:
Why do we need both of these? It seens like the first does the same thing as the second with `sign_extend` set to false. Why keep both?

Reply:
The difference is whether `pos` is in a register or not. The version where `pos` is an immediate value is used in many places to extract the mantissa  and exponent bits of floating point value; and version where `pos` is in a register is used in `code-generator-riscv64.cc` in `ASSEMBLE_ATOMIC_BINOP_EXT` etc. So we need both versions.